src/commands/app/upgrade.tsx

```diff
@@ -227,3 +274,61 @@ async function forceTargetVersionSelection(
     targetAppVersionString,
   );
 }
+
+async function updateMissingSystemSoftwareDependency(
+  process: ProcessRenderer,
+  apiClient: MittwaldAPIV2Client,
+  appInstallationId: string,
+  dependency: AppSystemSoftwareDependency,
+  force: boolean,
+) {
+  const dependencySoftware = await apiClient.app.getSystemsoftware({
+    systemSoftwareId: dependency.systemSoftwareId,
+  });
+  assertSuccess(dependencySoftware);
+
+  const dependencyVersionList = await apiClient.app.listSystemsoftwareversions({
+    systemSoftwareId: dependency.systemSoftwareId,
+  });
+  assertSuccess(dependencyVersionList);
+
+  let dependencyTargetVersion: AppSystemSoftwareVersion = {
+    id: "not yet set",
```

Comment:
Maybe add an assertion after the `for` loop that a target version was actually set (considering the case that the loop was completed with no version matching the selector)?